### PR TITLE
Add support and tests for semicontinuous and semi integer variables

### PR DESCRIPTION
**Pull request by rafabench**
Solved issue #61 
The tests `SemiContInt\semicontint.jl` are not available on MOI.Test yet.

### CODE REVIEW

File: src/MOI/MOI_wrapper.jl [hunk 1]
```diff
@@ -1117,6 +1125,13 @@ function _get_variable_lower_bound(model, info)
     return lb == -Xpress.Lib.XPRS_MINUSINFINITY ? -Inf : lb
 end
 
+function _get_variable_semi_lower_bound(model, info)
+    if !isnan(info.semi_lower_bound)
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl]: this is equivale to just returning the value :)

File: src/MOI/MOI_wrapper.jl [hunk 2]
```diff
@@ -1300,7 +1315,9 @@ function MOI.add_constraint(
     _throw_if_existing_lower(info.bound, info.type, typeof(s), f.variable)
     _throw_if_existing_upper(info.bound, info.type, typeof(s), f.variable)
     Xpress.chgcoltype(model.inner, [info.column], Cchar['S'])
-    _set_variable_lower_bound(model, info, s.lower)
+    _set_variable_lower_bound(model, info, 0.0)
+    Xpress.chgglblimit(model.inner, [info.column], Float64[s.lower])
+    info.semi_lower_bound = s.lower
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl]: use _set_variable_semi_lower_bound here

File: src/MOI/MOI_wrapper.jl [hunk 3]
```diff
@@ -1098,6 +1100,12 @@ function _set_variable_lower_bound(model, info, value)
     # end
 end
 
+function _set_variable_semi_lower_bound(model, info, value)
+    info.semi_lower_bound = value
+    _set_variable_lower_bound(model, info, 0.0)
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl]: not sure about having this implicitly here. maybe add docstrings

File: src/MOI/MOI_wrapper.jl [hunk 4]
```diff
@@ -1339,8 +1357,10 @@ function MOI.add_constraint(
     info = _info(model, f.variable)
     _throw_if_existing_lower(info.bound, info.type, typeof(s), f.variable)
     _throw_if_existing_upper(info.bound, info.type, typeof(s), f.variable)
-    Xpress.chgcoltype(model.inner, [info.column], Cchar['P'])
-    _set_variable_lower_bound(model, info, s.lower)
+    Xpress.chgcoltype(model.inner, [info.column], Cchar['R'])
+    _set_variable_lower_bound(model, info, 0.0)
+    info.semi_lower_bound = s.lower
+    Xpress.chgglblimit(model.inner, [info.column], Float64[s.lower])
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl]: again, user set semi lower

File: src/MOI/MOI_wrapper.jl [hunk 5]
```diff
@@ -84,7 +85,8 @@ mutable struct VariableInfo
             "",
             "",
             NaN,
-            0
+            0,
+            NaN
```
Comment (joaquimg) [inline src/MOI/MOI_wrapper.jl]: there is something off about the number of arguments that is no matching with the number of attributes

Comment (rafabench) [reply src/MOI/MOI_wrapper.jl]: The attributes previous_lower_bound and previous_upper_bound weren't initialized in the arguments of the constructor. But this problem is still on the master branch

Comment (joaquimg) [reply src/MOI/MOI_wrapper.jl]: got it! good that you fixed.

**Pull request merged**